web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -262,22 +256,22 @@ export function EnrollEksCluster(props: AgentStepProps) {
     !selectedCluster ||
     enrollmentState.status !== 'notStarted';
 
-  let command = '';
+  let manualCommandProps: GenerateCmdProps = null;
   if (selectedCluster) {
-    command = generateCmd({
+    manualCommandProps = {
       namespace: 'teleport-agent',
       clusterName: selectedCluster.name,
       proxyAddr: ctx.storeUser.state.cluster.publicURL,
-      tokenId: joinToken.id,
       clusterVersion: ctx.storeUser.state.cluster.authVersion,
-      resourceId: joinToken.internalResourceId,
+      tokenId: '', // Filled in by the ManualHelmDialog.
+      resourceId: '',
```

Comment:
Since `manualCommandProps` are used only in `ManualHelmDialog` and whenever we set the token, we also `generateCmd`, another approach we could take is to create here a function called something like `setJoinTokenAndGenerateCmd` and pass it to `ManualHelmDialog`. This sort-of leads to better ergonomics (you just need to pass one prop vs an object with many fields; you don't leave holes in data structures that need to be filled in) and it also encapsulates the business logic a little bit better since it's easier to see how and when `joinToken` gets set.

But I don't think the current approach is bad, I just wanted to show an alternative approach.

Reply:
Yeah, I was never a fan of sending all the information into the dialog. Changed to the way you suggested https://github.com/gravitational/teleport/pull/37730/commits/0d9a1584f0b99545f560d9b8c7f1bcbf59316ca9